qt/scientific_interfaces/Inelastic/BayesFitting/StretchView.h

```diff
@@ -0,0 +1,121 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "DllConfig.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/DoubleEditorFactory.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/QtTreePropertyBrowser"
+#include "MantidQtWidgets/Common/UserInputValidator.h"
+#include "MantidQtWidgets/Spectroscopy/RunWidget/IRunSubscriber.h"
+#include "StretchData.h"
+#include "ui_Stretch.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+class MANTIDQT_INELASTIC_DLL IStretchViewSubscriber {
```

Comment:
This interface class might need a virtual default destructor to ensure objects that inherit from it can destruct this part of the object